gevent/_socket2.py

```diff
@@ -334,10 +334,10 @@ def settimeout(self, howlong):
             howlong = f()
             if howlong < 0.0:
                 raise ValueError('Timeout value out of range')
-        self.timeout = howlong
+        self.__dict__['timeout'] = howlong # avoid recursion with any property on self.timeout
```

Comment:
I don't remember specifically which version, but the later comment setting up the property indicates that at least one of the CPython builds I was testing this on, under some circumstances at least, failed to have a 'timeout' value, so that line raised AttributeValue. (It always raises under PyPy.) A quick check shows it to be missing from 2.6 and 2.7, but present in >= 3.2. I *think* the error condition depended on whether a timeout was already set or not, but without rerunning the full test matrix I can't be sure.